lib/ember-qunit/qunit-module.js

```diff
@@ -45,21 +35,24 @@ export function createModule(Constructor, name, description, callbacks) {
   var module = new Constructor(name, description, callbacks);
 
   qunitModule(module.name, {
-    setup: function(assert) {
-      var done = assert.async();
-      return module.setup().then(function() {
+    beforeEach() {
+      return module.setup(...arguments).then(() => {
+        Object.keys(module.context).forEach((key) => {
+          this[key] = module.context[key];
+        });
+
         if (beforeEach) {
-          beforeEach.call(module.context, assert);
+          beforeEach.apply(this, arguments);
```

Comment:
So after thinking about this more, I think I'd prefer keeping it as `this`, just because that's what is going to happen for the `test` callback. Thoughts? In the end, it's the same reference so I don't feel super strongly, but I think it makes it a bit clearer as to the expected behavior.